L1Trigger/L1TTrackMatch/plugins/L1TkElectronTrackProducer.cc

```diff
@@ -0,0 +1,340 @@
+// -*- C++ -*-
+//
+/**\class L1TkElectronTrackMatchAlgo
+
+ Description: Producer of a TkElectron, for the algorithm matching a L1Track to the L1EG object
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/L1TCorrelator/interface/TkElectron.h"
+#include "DataFormats/L1TCorrelator/interface/TkElectronFwd.h"
+
+#include "DataFormats/Math/interface/LorentzVector.h"
+
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+
+// Matching Algorithm
+#include "L1Trigger/L1TTrackMatch/interface/L1TkElectronTrackMatchAlgo.h"
+#include "L1Trigger/L1TTrackMatch/interface/L1TkElectronEtComparator.h"
+#include "L1Trigger/L1TTrackMatch/interface/pTFrom2Stubs.h"
+
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "DataFormats/L1Trigger/interface/EGamma.h"
+
+#include <string>
+#include "TMath.h"
+
+using namespace l1t;
+
+//
+// class declaration
+//
+
+class L1TkElectronTrackProducer : public edm::EDProducer {
+public:
+  typedef TTTrack<Ref_Phase2TrackerDigi_> L1TTTrackType;
+  typedef std::vector<L1TTTrackType> L1TTTrackCollectionType;
+
+  explicit L1TkElectronTrackProducer(const edm::ParameterSet&);
+  ~L1TkElectronTrackProducer() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void beginJob() override;
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
+
+  //virtual void beginRun(edm::Run&, edm::EventSetup const&);
+  //virtual void endRun(edm::Run&, edm::EventSetup const&);
+  //virtual void beginLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&);
+  //virtual void endLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&);
+
+  float isolation(const edm::Handle<L1TTTrackCollectionType>& trkHandle, int match_index);
+  double getPtScaledCut(double pt, std::vector<double>& parameters);
+  bool selectMatchedTrack(double& d_r, double& d_phi, double& d_eta, double& tk_pt, float& eg_eta);
+
+  // ----------member data ---------------------------
+  std::string label;
+
+  float ETmin;  // min ET in GeV of L1EG objects
```

Comment:
member variables should start with lowercase and end in trailing underscore (code rules)

Reply:
done